pyartifactory/models/repository.py

```diff
@@ -306,3 +312,54 @@ class RemoteRepositoryResponse(RemoteRepository):
     enableVagrantSupport: bool = False
     enableGitLfsSupport: bool = False
     enableDistRepoSupport: bool = False
+
+
+class FederatedRepository(BaseRepositoryModel):
+    """Models a federated repository."""
+
+    rclass: Literal[RClassEnum.federated] = RClassEnum.federated
+    members: List[FederatedMembers] = []
+    checksumPolicyType: ChecksumPolicyType = ChecksumPolicyType.client_checksums
+    handleReleases: bool = True
+    handleSnapshots: bool = True
+    maxUniqueSnapshots: int = 0
+    maxUniqueTags: int = 0
+    debianTrivialLayout: bool = False
+    snapshotVersionBehavior: SnapshotVersionBehavior = SnapshotVersionBehavior.non_unique
+    suppressPomConsistencyChecks: bool = False
+    blackedOut: bool = False
+    xrayIndex: bool = False
+    propertySets: Optional[List[str]] = None
+    dockerApiVersion: str = "V2"
+    archiveBrowsingEnabled: bool = False
+    calculateYumMetadata: bool = False
+    yumRootDepth: int = 0
+    enableFileListsIndexing: bool = False
+    optionalIndexCompressionFormats: Optional[List[str]] = None
+    downloadRedirect: bool = False
+    cdnRedirect: bool = False
+    blockPushingSchema1: bool = False
+    primaryKeyPairRef: Optional[str] = None
+    secondaryKeyPairRef: Optional[str] = None
+    priorityResolution: bool = False
+    cargoInternalIndex: bool = False

```

Comment:
If the `FederatedRepository` is just a LocalRepository but with some additional fields as you mentioned. Why not do something like this ?

```
class FederatedRepository(LocalRepository):
       rclass: Literal[RClassEnum.federated] = RClassEnum.federated
       members: List[FederatedMembers] = []
```

Same for FederatedRepositoryResponse:
```
class FederatedRepository(LocalRepositoryResponse):
       rclass: Literal[RClassEnum.federated] = RClassEnum.federated
       members: List[FederatedMembers] = []
```

Doing this would reduce a lot of code duplication